cli/src/semdep/parsers/composer.py

```diff
@@ -0,0 +1,121 @@
+"""
+Parsers for composer.lock files
+Based on https://getcomposer.org/doc/01-basic-usage.md
+"""
+# Import necessary modules and classes
```

Comment:
General advice, it's not super useful to write comments describing what is literally happening; we can read the code to see we're importing stuff. Comments are more useful for context/why something is being done, or high level descriptions of hard to read code. Something that may not be clear to a reader. Like "we do this in batches because it's slow" or "X data structure represents X idea".